setup/terraform/resources/cdsw_setup.py

```diff
@@ -22,172 +22,255 @@
+        'kernel': 'python3',
+        'cpu': 1,
+        'memory': 4,
+        'nvidia_gpu': 0,
+        'notifications': [{
+            'user_id': 1,
+            'success': False,
+            'failure': False,
+            'timeout': False,
+            'stopped': False
+        }],
+        'recipients': {},
+        'attachments': [],
+    })
+job_id = r.json()['id']
+print('Job ID: %s' % (job_id,))
+
+print('# Start job')
+job_url = '%s/projects/admin/edge2ai-workshop/jobs/%s' % (CDSW_API, job_id)
+start_url = '%s/start' % (job_url,)
+
+r = s.post(start_url, json={})
+
+while True:
+    r = s.get(job_url)
+    status = r.json()['latest']['status']
+    print('Job %s status: %s' % (job_id, status))
+    if status == 'succeeded':
+        break
+    elif status == 'failed':
+        raise RuntimeError('Job failed')
+    time.sleep(10)
+
+print('# Get engine image to use for model')
+r = s.get(CDSW_API + '/projects/admin/edge2ai-workshop/engine-images')
+engine_image_id = r.json()['id']
+print('Engine image ID: %s' % (engine_image_id,))
+
+print('# Deploy model')
+r = s.post(CDSW_ALTUS_API + '/models/create-model', json={
+        'projectId': project_id,
+        'name': model_name,
+        'description': model_name,
+        'visibility': 'private',
+        'targetFilePath': 'cdsw.iot_model.py',
+        'targetFunctionName': 'predict',
+        'engineImageId': engine_image_id,
+        'kernel': 'python3',
+        'examples': [{'request': {'feature': '0, 65, 0, 137, 21.95, 83, 19.42, 111, 9.4, 6, 3.43, 4'}}],
+        'cpuMillicores': 1000,
+        'memoryMb': 4096,
+        'replicationPolicy': {'type': 'fixed', 'numReplicas': 1},
+        'environment': {},
+    })
+model_id = r.json()['id']
+print('Model ID: %s' % (model_id,))
+
+# ================================================================================
+
+# See https://docs.cloudera.com/cdsw/latest/analytical-apps/topics/cdsw-application-limitations.html
+
+print('# Allow applications to be configured with unauthenticated access')
+r = s.patch(CDSW_API + '/site/config', json={"allow_unauthenticated_access_to_app": True})
+print('Set unauthenticated access flag to: %s'% (r.json()["allow_unauthenticated_access_to_app"],))
+
+print('# Add project for Data Visualization server')
+project_name = 'viz'
+viz_project_id = None
+r = s.get(CDSW_API + '/users/admin/projects')
+for project in r.json():
+    if project['name'] == project_name:
+        viz_project_id = project['id']
+        break
+if not viz_project_id:
+    r = s.post(CDSW_API + '/users/admin/projects', json={'template': 'blank',
+                                                         'project_visibility': 'private',
+                                                         'name': project_name})
+    viz_project_id = r.json()['id']
+print('Viz project ID: %s'% (viz_project_id,))
+
+print('# Add custom engine for Data Visualization server')
+#docker.repository.cloudera.com/cloudera/cdv/cdswdataviz:6.2.3-b18
```

Comment:
I don't think this comment is necessary

Reply:
Done. Need to test with a more recent Viz image and if everything works, will update the image tag in the script as well.